pkg/env/core.go

```diff
@@ -84,6 +84,15 @@ func NewCoreForCI(ctx context.Context, log *logrus.Entry) (Core, error) {
 	}, nil
 }
 
+func DeriveCoreForCluster(env Core) Core {
+	_core := env.(*core)
+
+	return &core{
+		InstanceMetadata: instancemetadata.NewClusterMetadata(_core.InstanceMetadata),
+		deploymentMode:   _core.deploymentMode,
+	}
+}
+
```

Comment:
If this is only used in Hacks helper to create cluster, can we keep it there? This confuses existing cores. It looks like it basically required some kind of "fake InstanceMetadata" with one field overriden.

I would not want to leak this into instancemetada code and code code :/ and all these renames `env` to `clusterEnv` adds even more confusion